Index: baresip-2.9.0/modules/webrtc_aec/module.mk
===================================================================
--- baresip-2.9.0.orig/modules/webrtc_aec/module.mk
+++ baresip-2.9.0/modules/webrtc_aec/module.mk
@@ -17,7 +17,7 @@ CPPFLAGS	+= -isystem $(WEBRTC_PATH) -fPI
 
 $(MOD)_LFLAGS	+= \
 	-L$(WEBRTC_PATH)/lib/x64/Debug \
-	-lwebrtc_audio_processing \
+	-lwebrtc-audio-processing-1 \
 	-lstdc++
 
 
