Index: re-2.9.0/include/resolv_compat.h
===================================================================
--- /dev/null
+++ re-2.9.0/include/resolv_compat.h
@@ -0,0 +1,29 @@
+#if !defined(__GLIBC__)
+/***************************************************************************
+ * resolv_compat.h
+ *
+ * Mimick GLIBC's res_ninit() and res_nclose() for musl libc
+ * Note: res_init() is actually deprecated according to
+ * http://docs.oracle.com/cd/E36784_01/html/E36875/res-nclose-3resolv.html
+ **************************************************************************/
+#include <string.h>
+
+static inline int res_ninit(res_state statp)
+{
+	int rc = res_init();
+	if (statp != &_res) {
+		memcpy(statp, &_res, sizeof(*statp));
+	}
+	return rc;
+}
+
+static inline int res_nclose(res_state statp)
+{
+	if (!statp)
+		return -1;
+	if (statp != &_res) {
+		memset(statp, 0, sizeof(*statp));
+	}
+	return 0;
+}
+#endif
Index: re-2.9.0/src/dns/res.c
===================================================================
--- re-2.9.0.orig/src/dns/res.c
+++ re-2.9.0/src/dns/res.c
@@ -9,6 +9,9 @@
 #include <netinet/in.h>
 #include <arpa/nameser.h>
 #include <resolv.h>
+#if defined(__linux__) && !defined(__GLIBC__)
+#include "resolv_compat.h"
+#endif
 #include <string.h>
 #include <re_types.h>
 #include <re_fmt.h>
