Index: webrtc-audio-processing-1.0/meson.build
===================================================================
--- webrtc-audio-processing-1.0.orig/meson.build
+++ webrtc-audio-processing-1.0/meson.build
@@ -4,7 +4,7 @@ project('webrtc-audio-processing-1', 'c'
   default_options : [ 'warning_level=1',
                       'buildtype=debugoptimized',
                       'c_std=c11',
-                      'cpp_std=c++14',
+                      'cpp_std=c++17',
                     ]
 )
 
@@ -46,7 +46,7 @@ absl_dep = [
   cpp.find_library('absl_flags_internal'),
   cpp.find_library('absl_flags_marshalling'),
   cpp.find_library('absl_flags_parse'),
-  cpp.find_library('absl_flags_registry'),
+  cpp.find_library('absl_flags_reflection'),
   cpp.find_library('absl_flags_usage_internal'),
   cpp.find_library('absl_raw_logging_internal'),
   cpp.find_library('absl_strings'),
